screen-releases/src/main/java/com/ivy/releases/ReleasesScreen.kt

```diff
@@ -0,0 +1,200 @@
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalUriHandler
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.ivy.navigation.Navigation
+import com.ivy.navigation.navigation
+
+@Composable
+fun ReleasesScreenImpl() {
+    val viewModel: ReleasesViewModel = viewModel()
+    val uiState = viewModel.uiState()
+
+    ReleasesUi(
+        uiState = uiState,
+        onEvent = {
+            viewModel.onEvent(it)
+        }
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun ReleasesUi(
+    uiState: ReleasesState,
+    onEvent: (ReleasesEvent) -> Unit
+) {
+    val nav = navigation()
+
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                title = {
+                    TopAppBarTitle(title = "Releases")
+                },
+                navigationIcon = {
+                    BackButton(nav = nav)
+                }
+            )
+        }
+    ) { innerPadding ->
+        LazyColumn(
+            modifier = Modifier.padding(start = 12.dp, end = 12.dp, bottom = 12.dp),
+            verticalArrangement = Arrangement.spacedBy(8.dp),
+            contentPadding = innerPadding,
+        ) {
+            content(
+                releasesState = uiState,
+                onEvent = onEvent
+            )
+        }
+    }
+}
+
+@Composable
+private fun TopAppBarTitle(title: String) {
+    Text(
+        text = title,
+        maxLines = 1,
+        overflow = TextOverflow.Ellipsis
+    )
+}
+
+@Composable
+private fun BackButton(nav: Navigation) {
+    IconButton(onClick = {
+        nav.back()
+    }) {
+        Icon(
+            imageVector = Icons.Filled.ArrowBack,
+            contentDescription = "Back"
+        )
+    }
+}
+
+private fun LazyListScope.content(
+    releasesState: ReleasesState,
+    onEvent: (ReleasesEvent) -> Unit
+) {
+    when (releasesState) {
+        is ReleasesState.Error -> item {
+            ReleasesErrorState(
+                message = releasesState.errorMessage,
+                onClick = {
+                    onEvent(ReleasesEvent.OnTryAgainClicked)
+                }
+            )
+        }
+
+        is ReleasesState.Loading -> item {
+            Text(text = releasesState.loadingMessage)
+        }
+
+        is ReleasesState.Success -> items(releasesState.releasesInfo) {
+            ReleaseInfoCard(releaseInfo = it)
+        }
+    }
+}
+
+@Composable
+private fun ReleaseInfoCard(
+    releaseInfo: ReleaseInfo,
+    modifier: Modifier = Modifier
+) {
+    val browser = LocalUriHandler.current
+
+    OutlinedCard(
+        modifier = modifier.clickable {
```

Comment:
Use `onClick = {` on the OutlinedCard so when clicking, the click area is clipped properly